inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/reduce/reduce_kernel_base.cpp

```diff
@@ -0,0 +1,278 @@
+/*
+// Copyright (c) 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "reduce_kernel_base.h"
+#include "kernel_selector_utils.h"
+#include <vector>
+#include <string>
+#include "common_tools.h"
+
+namespace kernel_selector {
+
+bool ReduceKernelBase::Validate(const Params& p, const optional_params&) const {
+    auto& params = dynamic_cast<const reduce_params&>(p);
+
+    if (params.GetType() != KernelType::REDUCE) {
+        return false;
+    }
+
+    for (auto& fused_op : params.fused_ops) {
+        if (!IsFusedPrimitiveSupported(fused_op))
+            return false;
+    }
+
+    return true;
+}
+
+CommonDispatchData ReduceKernelBase::SetDefault(const reduce_params& params, const optional_params&) const {
```

Comment:
I don't think that `SetDefault` implementation should be part of base. GWS and LWS calculations are highly implementation specific and I'd expect that most implementations will override it any way.